safety_test.go

```diff
@@ -23,107 +23,184 @@ import (
 func TestLimitedGroupCleanup(t *testing.T) {
 	t.Parallel()
 	var counter int64
+	var leak contextLeak
+
 	opsQueue := func() chan func(context.Context) {
 		g := Limited(context.Background(), 10)
 		for i := 0; i < 100; i++ {
-			g.Go(func(context.Context) {
+			g.Go(func(ctx context.Context) {
 				atomic.AddInt64(&counter, 1)
+				leak.leak(ctx)
 			})
 		}
 		return g.(*limitedGroup).ops
 		// leak the un-awaited group
 	}()
 	assert.NotNil(t, opsQueue)
 	runtime.GC() // Trigger cleanups for leaked resources
+	runtime.GC() // Trigger cleanups for leaked resources
+	runtime.GC() // Trigger cleanups for leaked resources
```

Comment:
what's the deal with the repeat?

Reply:
some of the structs that must be cleaned up in order to trigger the finalizers under test here are transitively owned, so we need to pump the handle a couple times to make them get cleaned up. the current GC won't dispose of everything that isn't transitively reachable in a single pass; if you have A -> B -> C and you forget about A, C may not get garbage collected until the third cycle.

if you remove the extra `GC()` calls in these tests so that there is only one, some of them *will* hang forever; mostly they work totally fine if you put two, so i put three just for good measure.

Reply:
there's also a tiny degree of randomness here i think